nifty-ssl/src/main/java/com/facebook/nifty/ssl/NiftyOpenSslServerContext.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (C) 2012-2013 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.nifty.ssl;
+
+import org.apache.tomcat.jni.Pool;
+import org.apache.tomcat.jni.SSL;
+import org.apache.tomcat.jni.SSLContext;
+import org.apache.tomcat.jni.SessionTicketKey;
+import org.jboss.netty.handler.ssl.OpenSslEngine;
+import org.jboss.netty.handler.ssl.SslBufferPool;
+import org.jboss.netty.handler.ssl.SslHandler;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A lot of this code is taken from Netty's OpenSslServerContext.
+ * NiftyOpenSslServerContext allows us to control this instantiation of SSLContext with custom options
+ * missing in netty. We should be able to get rid of this when we move the dependency
+ * to the latest version of netty.
```

Comment:
Please add a comment with the version number you copied it from. Just for future reference